### DISCUSSION TITLE
UnicodeEncodeError: 'ascii' codec can't encode character '\u202e' in position 14: ordinal not in range(128)

### DISCUSSION THREAD

**Discussion by AdonisPro** [mitmproxy/mitmproxy#5213] [Q&A]
H Team,

In connection to the issue #5211, i am re-logging the issue as Bug.

I am trying to run a python script which contains unicode character U202E. Whenever i am trying to run this script with mitmdump, it is throwing error: UnicodeEncodeError: 'ascii' codec can't encode character '\u202e' in position 6: ordinal not in range(128), when i am trying to perform any operation in target machine.

I have tried with mitmproxy version 2.0.2 and 8.0.0, 2.0.0, 0.15, 0.18 but same result everywhere. Also sometimes i see logs throwing error showing some other unicode characters like \u2026 etc.

I am putting my code (modified from opensource code available in github.) for your testing purpose and to reproduce the issue

#!/usr/bin/python3
file = 'abcd\u202Efdp.exe'
with open(file, 'w') as f:
    f.write('hello"')

![image](https://user-images.githubusercontent.com/91906895/160169056-fbf42f4d-fa84-46b5-b5ac-523fd98e7580.png)

In my actual script, this sort of operation is happening and mitmdump is failing there also. If you run the python code normally it does all its operation an it creates the file spoofing the extension.

![image](https://user-images.githubusercontent.com/91906895/160170822-fb2078a9-40f6-4158-9462-0fb4686b673b.png)

I'd request you to please look into this and help me by fixing it as in many rleases/version i have checked, i have found the same error everywhere.

**Comment by Prinzhorn** [comment]
addon.py

```py
#!/usr/bin/python3
file = "abcd\u202Efdp.exe"
with open(file, "w") as f:
    f.write('hello"')
```

```sh
 $ ~/Downloads/mitmproxy-8.0.0-linux/mitmdump -s addon.py 
Loading script addon.py
Proxy server listening at http://*:8080
```

```
 $ ll
-rw-rw-r--  1 alex alex     6 Mär 25 18:33 abcd‮fdp.exe
-rw-rw-r--  1 alex alex    95 Mär 25 18:30 addon.py
```

For me it works as expected and the file is created.

Please follow the issue template when reporting bugs. For example you are missing the "System Information" part.

**Comment by AdonisPro** [comment]
Sorry , i missed out sys info:

![image](https://user-images.githubusercontent.com/91906895/160174010-0d021a35-4d55-4e18-b0e8-2745abb0b9e9.png)
![image](https://user-images.githubusercontent.com/91906895/160174073-8ab9f563-68ef-4410-939d-59434758faad.png)

But i am constantly facing this unicode error. Please suggest.

**Comment by mhils** [comment]
What output do you get if you put 
```python
import sys
print(sys.getdefaultencoding())
```
in your addon?
